substrate-archive/src/actors.rs

```diff
@@ -42,30 +39,32 @@ use sp_runtime::traits::{Block as BlockT, Header as _, NumberFor};
 use std::marker::PhantomData;
 use std::panic::AssertUnwindSafe;
 use std::sync::Arc;
+use substrate_archive_backend::{ApiAccess, Meta, ReadOnlyBackend};
+pub use substrate_archive_common::{msg, msg::Die, Result, database::ReadOnlyDB};
 use xtra::prelude::*;
 
 // TODO: Split this up into two objects
 // System should be a factory that produces objects that should be spawned
 
 /// Context that every actor may use
 #[derive(Clone)]
-pub struct ActorContext<B: BlockT + Unpin>
+pub struct ActorContext<B: BlockT + Unpin, D: ReadOnlyDB + 'static>
```

Comment:
it looks like you need to add `Clone` bound in order for this to work after changing to `static dispatch` because it is `cloned` in the Archive impl.